src/libsyntax_ext/proc_macro_registrar.rs

```diff
@@ -215,6 +224,29 @@ impl<'a> CollectProcMacros<'a> {
             self.handler.span_err(item.span, msg);
         }
     }
+
+    fn collect_bang_proc_macro(&mut self, item: &'a ast::Item, attr: &'a ast::Attribute) {
+        if let Some(_) = attr.meta_item_list() {
+            self.handler.span_err(attr.span, "`#[proc_macro]` attribute
+                cannot contain any meta items");
```

Comment:
"meta item" is compiler jargon -- I'd say "does not accept arguments" (also applies to `proc_macro_attribute`).